common/src/test/java/org/conscrypt/java/security/KeyFactoryTestEC.java

```diff
@@ -15,54 +15,129 @@
  */
 package org.conscrypt.java.security;
 
+import static org.junit.Assert.fail;
+
+import java.math.BigInteger;
+import java.security.KeyFactory;
 import java.security.KeyPair;
 import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.security.Security;
 import java.security.interfaces.ECPrivateKey;
 import java.security.interfaces.ECPublicKey;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
 import java.security.spec.ECPrivateKeySpec;
 import java.security.spec.ECPublicKeySpec;
 import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
 import java.util.Arrays;
 import java.util.List;
+import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 import tests.util.ServiceTester;
 
 @RunWith(JUnit4.class)
-public class KeyFactoryTestEC extends
-    AbstractKeyFactoryTest<ECPublicKeySpec, ECPrivateKeySpec> {
-
-  public KeyFactoryTestEC() {
-    super("EC", ECPublicKeySpec.class, ECPrivateKeySpec.class);
-  }
-
-  @Override
-  public ServiceTester customizeTester(ServiceTester tester) {
-    // BC's EC keys always use explicit params, even though it's a bad idea, and we don't support
-    // those, so don't test BC keys
-    return tester.skipProvider("BC");
-  }
-
-  @Override
-  protected void check(KeyPair keyPair) throws Exception {
-    new SignatureHelper("SHA256withECDSA").test(keyPair);
-  }
-
-  @Override
-  protected List<KeyPair> getKeys() throws NoSuchAlgorithmException, InvalidKeySpecException {
-    return Arrays.asList(
-        new KeyPair(
-            DefaultKeys.getPublicKey("EC"),
-            DefaultKeys.getPrivateKey("EC")
-        ),
-        new KeyPair(
-            new TestPublicKey(DefaultKeys.getPublicKey("EC")),
-            new TestPrivateKey(DefaultKeys.getPrivateKey("EC"))
-        ),
-        new KeyPair(
-            new TestECPublicKey((ECPublicKey)DefaultKeys.getPublicKey("EC")),
-            new TestECPrivateKey((ECPrivateKey)DefaultKeys.getPrivateKey("EC"))
-        )
-    );
-  }
+public class KeyFactoryTestEC extends AbstractKeyFactoryTest<ECPublicKeySpec, ECPrivateKeySpec> {
+    public KeyFactoryTestEC() {
```

Comment:
nit: Ideally don't reformat existing Conscrypt code which doesn't follow the Android coding style as it makes changes harder to track via "blame".  As these are fairly minor test classes, don't worry about it in this CL though.